src/main/java/com/flippingutilities/jobs/SlotStateSenderJob.java

```diff
@@ -0,0 +1,160 @@
+package com.flippingutilities.jobs;
+
+import com.flippingutilities.controller.FlippingPlugin;
+import com.flippingutilities.model.OfferEvent;
+import com.flippingutilities.ui.widgets.SlotActivityTimer;
+import com.flippingutilities.utilities.SlotState;
+import com.flippingutilities.utilities.SlotsUpdate;
+import com.flippingutilities.utilities.WikiRequest;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.GrandExchangeOffer;
+import net.runelite.api.events.GrandExchangeOfferChanged;
+import okhttp3.*;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.*;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+@Slf4j
+public class SlotStateSenderJob {
+    FlippingPlugin plugin;
+    ScheduledExecutorService executor;
+    OkHttpClient httpClient;
+    Future slotStateSenderTask;
+    SlotsUpdate previouslySentSlotUpdate;
+    List<Consumer<Integer>> subscribers = new ArrayList<>();
+    public static int PERIOD = 10; //seconds
+    public boolean justLoggedIn = false;
+
+    public SlotStateSenderJob(FlippingPlugin plugin, OkHttpClient httpClient) {
+        this.plugin = plugin;
+        this.httpClient = httpClient;
+        this.executor = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    public void subscribe(Consumer<Integer> subscriber) {
+        subscribers.add(subscriber);
+    }
+
+    public void start() {
+        slotStateSenderTask = executor.scheduleAtFixedRate(this::sendSlots, 10, PERIOD, TimeUnit.SECONDS);
+        log.info("started slot sender job");
+    }
+
+    public void stop() {
+        if (!slotStateSenderTask.isCancelled() && !slotStateSenderTask.isCancelled()) {
+            slotStateSenderTask.cancel(true);
+            log.info("shut down slot sender job");
+        }
+    }
+
+    private void sendSlots() {
+        if (!plugin.getApiAuthHandler().canCommunicateWithApi(plugin.getCurrentlyLoggedInAccount())) {
+            return;
+        }
+
+        List<SlotState> currentSlotStates = this.getCurrentSlots();
+        SlotsUpdate slotsUpdate = new SlotsUpdate(plugin.getCurrentlyLoggedInAccount(), currentSlotStates);
+        if (slotsUpdate.equals(this.previouslySentSlotUpdate) && !justLoggedIn) {
+            log.info("no updates to slots since the last time I sent them, not sending any requests.");
+            subscribers.forEach(subscriber -> subscriber.accept(0));
+            return;
+        }
+        justLoggedIn = false;
```

Comment:
While this name is fine, it's still a bit confusing
I would use something very explicit: `hasSentSlotStateSinceLogin`